### PR TITLE
Add seaborn

### PR DESCRIPTION
**Pull request by croth1**
picking up #721.

### CODE REVIEW

**Comment by croth1** [comment]
```
conda.resolve.NoPackagesFound: Package missing in current linux-64 channels: 
  - numpy 1.0*

./scripts/run_docker_build.sh returned exit code 1
```

Uh, what's wrong here? Why it is even trying to build against 1.0*?

**Comment by msarahan** [comment]
Random guess here, but you might need to use ```conda smithy rerender``` to get the CI build scripts updated.  I think that the CONDA_NPY variable defined in those should be what specifies the version of numpy that gets used.

**Comment by ocefpaf** [comment]
>
> Random guess here, but you might need to use conda smithy rerender to get
> the CI build scripts updated. I think that the CONDA_NPY variable defined
> in those should be what specifies the version of numpy that gets used.
>

Mike, there is no rerender on staged recipes. I am on my way home and tmr I
will take a look into the issue. (It is hard to debug this using only the
phone.)

**Comment by msarahan** [comment]
Thanks @ocefpaf, I guess the equivalent is probably a ```git pull --rebase origin master```?

File: recipes/seaborn/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - numpy
+    - matplotlib
```
Comment (jakirkham) [inline recipes/seaborn/meta.yaml]: Why are these 2 needed as build dependencies? Are we linking against them? Do they get imported at build time? Some combination?

File: recipes/seaborn/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - statsmodels
+    - patsy
+
+test:
+  imports:
+    - {{ name }}
+
+about:
+  home: http://stanford.edu/~mwaskom/software/seaborn/
+  license: BSD
```
Comment (jakirkham) [inline recipes/seaborn/meta.yaml]: Please specify clauses.

Comment (jakirkham) [inline recipes/seaborn/meta.yaml]: e.g. `BSD 3-Clause` (if it is 3-Clause)

**Comment by jakirkham** [comment]
Could this actually be coming from `conda-build-all`? Not to say it is necessarily even a `conda-build-all` problem at the heart of it. Please see [line]( https://github.com/SciTools/conda-build-all/blob/459814d56b68648d0a6164558312df4b5fc45e5c/conda_build_all/resolved_distribution.py#L29 ). It is pretty strange though as we did not use `x.x`. Would be curious to see if the problem persists if `numpy` and `matplotlib` are dropped from `build`.

**Comment by jakirkham** [comment]
Went ahead and tried the above suggestion and it appears that "fixes" the problem. Placed these changes in PR ( https://github.com/croth1/staged-recipes/pull/1 ), which should go right into this PR when merged. I'm very suspicious ATM that `conda-build` may be using `CONDA_NPY` when it shouldn't be. The reason being `conda-build-all` is the only thing I'm aware of that is setting [`CONDA_NPY`]( https://github.com/SciTools/conda-build-all/blob/459814d56b68648d0a6164558312df4b5fc45e5c/conda_build_all/resolved_distribution.py#L29 ) (particularly to `1.0`). However, that value shouldn't actually be used for anything real AFAIK and will actually be replaced if there is a real `numpy` version matrix to consider. Any ideas why this would happen, @msarahan?

**Comment by jakirkham** [comment]
Any interest in adding `patsy` too, @croth1? Could try starting by using commit ( https://github.com/conda-forge/staged-recipes/pull/1115/commits/61f40a83b63531aabe9574014fbe4961a9a0d52f ).

**Comment by croth1** [comment]
After rebasing my changes upon master the problem seems to be gone. Thanks all for your help here.

**Comment by croth1** [comment]
@jakirkham working on patsy right now.

**Comment by jakirkham** [comment]
Thanks @croth1. SGTM.

File: recipes/seaborn/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    - statsmodels
+    - patsy
+
+test:
+  imports:
+    - {{ name }}
+
+about:
+  home: http://stanford.edu/~mwaskom/software/seaborn/
+  license: BSD 3-Clause
```
Comment (jakirkham) [inline recipes/seaborn/meta.yaml]: Please add the license file too. This can be done as follows.

```
  license_file: LICENSE
```

Comment (jakirkham) [inline recipes/seaborn/meta.yaml]: Done below.

**Comment by jakirkham** [comment]
Forgot to raise the [NumPy 1.0 problem]( https://github.com/conda-forge/staged-recipes/pull/1252#issuecomment-239678584 ) so have now done so with `conda-build-all` issue ( https://github.com/SciTools/conda-build-all/issues/56 ) . Though maybe this is a `conda-build` issue in the end.

**Pull request merged**